### PR TITLE
project finished

### PR DESCRIPTION
**Pull request by Jinboi**
Project finished

As always, thank you for your guidance!

### CODE REVIEW

File: ExerciseTracker.Bata13/STUDY.ConsoleProjects.ExerciseTrackerFour/Data/ExerciseDbContext.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using STUDY.ConsoleProjects.ExerciseTrackerFour.Models;
+
+namespace STUDY.ConsoleProjects.ExerciseTrackerFour.Data;
+internal class ExerciseDbContext : DbContext
+{
+    public DbSet<Exercise> Exercises { get; set; }
+    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+    {
+        var configuration = new ConfigurationBuilder()
+            .AddJsonFile("appsettings.config")
```
Comment (stevietv) [inline ExerciseTracker.Bata13/STUDY.ConsoleProjects.ExerciseTrackerFour/Data/ExerciseDbContext.cs:12]: your code is looking for an `appsettings.config` json file, which doesn't exist.

**Review by stevietv — CHANGES_REQUESTED** [review]
Hi @Jinboi 
Currently your program crashes. Please fix this and check that it runs , then push up the fix.
Cheers!

File: ExerciseTracker.Bata13/STUDY.ConsoleProjects.ExerciseTrackerFour/Data/ExerciseDbContext.cs (hunk 1 above)
Comment (Jinboi) [reply ExerciseTracker.Bata13/STUDY.ConsoleProjects.ExerciseTrackerFour/Data/ExerciseDbContext.cs]: I've fixed the bug and re-submitted.
Thank you!

**Comment by stevietv** [comment]
Hi @Jinboi 
thanks for the update. I've ran your program and encountered a few issues:

* The program crashes when I enter an invalid date. It seems you are only using TryParseExact in the update method but not the add method.
* you have not used Dependency Injection, which is a requirement of the project

Can you take a look at working on meeting the requirement for dependency injection

**Comment by Jinboi** [comment]
Hi Stevietvm

I've re-submitted after fixing DI request and invalid bug. Please let me know if there's anything wrong! Thank you!

**Comment by TheCSharpAcademy** [comment]
@stevietv Any updates on this one? 🤓

**Comment by stevietv** [comment]
@TheCSharpAcademy  - see my discord message asking for your input

**Comment by stevietv** [comment]
I discussed with Pablo and he was happy with the usage of DI so good work! ✅

**Review by stevietv — APPROVED** [review]

**Pull request merged**